iroh/src/client/blobs.rs

```diff
@@ -1248,4 +1251,185 @@ mod tests {
+                    tag: SetTagOption::Auto,
+                    mode: DownloadMode::Queued,
+                },
+            )
+            .await?
+            .await;
+        assert!(res.is_err());
+        assert_eq!(
+            res.err().unwrap().to_string().as_str(),
+            "No provider nodes found"
+        );
+
+        Ok(())
+    }
+
+    /// Download a existing collection. Check that things succeed and no download is performed.
+    #[tokio::test]
+    async fn test_blob_get_existing_collection() -> TestResult<()> {
+        let _guard = iroh_test::logging::setup();
+
+        let node = crate::node::Node::memory().spawn().await?;
+        // We use a nonexisting node id because we just want to check that this succeeds without
+        // hitting the network.
+        let node_id = NodeId::from_bytes(&[0u8; 32])?;
+        let client = node.client();
+
+        let mut collection = Collection::default();
+        let mut tags = Vec::new();
+        let mut size = 0;
+        for value in ["iroh", "is", "cool"] {
+            let import_outcome = client.blobs().add_bytes(value).await.context("add bytes")?;
+            collection.push(value.to_string(), import_outcome.hash);
+            tags.push(import_outcome.tag);
+            size += import_outcome.size;
+        }
+
+        let (hash, _tag) = client
+            .blobs()
+            .create_collection(collection, SetTagOption::Auto, tags)
+            .await?;
+
+        // load the hashseq and collection header manually to calculate our expected size
+        let hashseq_bytes = client.blobs().read_to_bytes(hash).await?;
+        size += hashseq_bytes.len() as u64;
+        let hashseq = HashSeq::try_from(hashseq_bytes)?;
+        let collection_header_bytes = client
+            .blobs()
+            .read_to_bytes(hashseq.into_iter().next().expect("header to exist"))
+            .await?;
+        size += collection_header_bytes.len() as u64;
+
+        // Direct
+        let res = client
+            .blobs()
+            .download_with_opts(
+                hash,
+                DownloadOptions {
+                    format: BlobFormat::HashSeq,
+                    nodes: vec![node_id.into()],
+                    tag: SetTagOption::Auto,
+                    mode: DownloadMode::Direct,
+                },
+            )
+            .await?
+            .await
+            .context("direct (download)")?;
+
+        assert_eq!(res.local_size, size);
+        assert_eq!(res.downloaded_size, 0);
+
+        // Queued
+        let res = client
+            .blobs()
+            .download_with_opts(
+                hash,
+                DownloadOptions {
+                    format: BlobFormat::HashSeq,
+                    nodes: vec![node_id.into()],
+                    tag: SetTagOption::Auto,
+                    mode: DownloadMode::Queued,
+                },
+            )
+            .await?
+            .await
```

Comment:
we should probably check the events in these tests, to make sure they look like we expect them, especially in the collection case